include/RAJA/util/View.hpp

```diff
@@ -296,6 +297,82 @@ RAJA_INLINE AtomicViewWrapper<ViewType, AtomicPolicy> make_atomic_view(
   return RAJA::AtomicViewWrapper<ViewType, AtomicPolicy>(view);
 }
 
+struct layout_left{};
+struct layout_right{};
+
+template<typename LAYOUT>
+struct Reshape;
+
+template<typename LAYOUT>
+struct Reshape;
+
+template<typename T>
+constexpr auto get_last_index(T last)
+{
+  return last;
+}
+
+template<typename T, typename... Args>
+constexpr auto get_last_index(T , Args... args)
+{
+  return get_last_index(args...);
+}
+
+template<std::size_t...Is>
+struct Reshape<std::index_sequence<Is...>>
+{
+  template<typename T, typename...Ts>
+  static auto get(T *ptr, Ts... s)
+  {
+    constexpr int N = sizeof...(Ts);
+    std::array<RAJA::idx_t, N> extent{s...};
+
+    auto custom_layout =
+      RAJA::make_permuted_layout(extent, std::array<RAJA::idx_t, N>{Is...});
+
+    constexpr auto unit_stride = get_last_index(Is...);
+
+    return RAJA::View<T, RAJA::Layout<N, RAJA::Index_type, unit_stride>>
+      (ptr, custom_layout);
+  }
+};
+
+template<>
+struct Reshape<layout_right>
+{
+  template<typename T, typename...Ts>
+  static auto get(T *ptr, Ts... s)
+  {
+    constexpr int N = sizeof...(Ts);
+    using view_t = RAJA::View<T, RAJA::Layout<N, RAJA::Index_type, N-1>>;
```

Comment:
In general spacing and when aliases or temporaries are used could be more consistent.

Reply:
yes, good call. I just hit with a focused version of clang-format and opted for using aliases when the view types became long.